here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/TagFilterHandler.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2017-2023 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+package com.here.naksha.lib.handlers;
+
+import static com.here.naksha.lib.core.util.CollectionUtils.isNotNullOrEmpty;
+import static com.here.naksha.lib.core.util.CollectionUtils.isNullOrEmpty;
+import static com.here.naksha.lib.handlers.AbstractEventHandler.EventProcessingStrategy.PROCESS;
+import static com.here.naksha.lib.handlers.AbstractEventHandler.EventProcessingStrategy.SEND_UPSTREAM_WITHOUT_PROCESSING;
+
+import com.here.naksha.lib.core.IEvent;
+import com.here.naksha.lib.core.INaksha;
+import com.here.naksha.lib.core.models.geojson.implementation.XyzFeature;
+import com.here.naksha.lib.core.models.geojson.implementation.namespaces.XyzNamespace;
+import com.here.naksha.lib.core.models.naksha.EventHandler;
+import com.here.naksha.lib.core.models.naksha.EventTarget;
+import com.here.naksha.lib.core.models.storage.*;
+import com.here.naksha.lib.core.util.json.JsonSerializable;
+import java.util.*;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TagFilterHandler extends AbstractEventHandler {
+
+  private static final Logger logger = LoggerFactory.getLogger(TagFilterHandler.class);
+
+  private @NotNull EventHandler eventHandler;
+  private @NotNull EventTarget<?> eventTarget;
```

Comment:
These two properties (`eventHandler`, `eventTarget`) are unused - do we need to declare them?
If it done due to constructor expectation from `PluginCache` then I would opt for refactoring of `PluginCache` so that it will allow different argument sets.

(btw I looked through and we have the same "issue" in `SourceIdHandler` where non of properties is used - maybe it's temporal 🤷 )

Reply:
Ok, will fix both handlers to avoid redundant variables that are not used currently.
For PluginCache - new backlog ticket MCPODS-7103